tools/load-testing/load-gas-loop.ts

```diff
@@ -0,0 +1,141 @@
+
+const printAddressInfo = async (address: string) => {
+  const nonce = await web3.eth.getTransactionCount(address);
+  const balance = await web3.eth.getBalance(address);
+  console.log(`Account: ${address} (nonce: ${nonce}) => ${balance}`);
+};
+
+const deployContract = async () => {
+  // 1M gas contract call (big_loop)
+
+  const code = await customRequest("eth_getCode", [contractAddress]);
+  if (code.result != "0x") {
+    console.log("Contract already deployed");
+    return;
+  }
+
+  const tx = await web3.eth.accounts.signTransaction(
+    {
+      from: deployer.address,
+      data: contractBytecode,
+      value: "0x00",
+      gasPrice: "0x00",
+      gas: "0x1000000",
+      nonce: 0,
+    },
+    deployer.privateKey
+  );
+  const result = await customRequest("eth_sendRawTransaction", [tx.rawTransaction]);
+  if (result.error) {
+    console.error(`Error deploying contract!`);
+    console.error(result.error);
+    return;
+  }
+  console.log(`Transaction sent: ${tx.transactionHash}`);
+  const startTime = Date.now();
+  while (Date.now() - startTime < 40000) {
+    let rcpt: TransactionReceipt = await web3.eth.getTransactionReceipt(tx.transactionHash);
+    if (rcpt) {
+      console.log(`Transaction done - block #${rcpt.blockNumber} (${rcpt.blockHash})`);
+      return;
+    }
+    await new Promise((resolve) => {
+      setTimeout(resolve, 2000);
+    });
+  }
+  throw new Error("Failed to verify contract deployment (timeout)");
+};
+
+const callContract = async (loopCount: number) => {
+  const contract = new web3.eth.Contract(contractAbi, contractAddress);
+
+  const encoded = await contract.methods.big_loop(loopCount).encodeABI();
+
+  const freshAccount = web3.eth.accounts.create();
+  const tx = await web3.eth.accounts.signTransaction(
+    {
+      from: freshAccount.address,
+      to: contractAddress,
+      data: encoded,
+      gasPrice: 0,
+      gas: "9627370496",
+      nonce: 0,
+    },
+    freshAccount.privateKey
+  );
+
+  const result = await customRequest("eth_sendRawTransaction", [tx.rawTransaction]);
+  if (result.error) {
+    console.error(result.error);
+    throw new Error(`Error calling contract!`);
+  }
+
+  console.log(`Transaction for Loop count ${loopCount} sent: ${tx.transactionHash}`);
+  const startTime = Date.now();
+  while (Date.now() - startTime < 60000) {
+    let rcpt: TransactionReceipt = await web3.eth.getTransactionReceipt(tx.transactionHash);
+    if (rcpt) {
+      console.log(`Loop count ${loopCount} - block #${rcpt.blockNumber} (${rcpt.blockHash})`);
+      return;
+    }
+    await new Promise((resolve) => {
+      setTimeout(resolve, 2000);
+    });
+  }
+  throw new Error("Failed to verify contract call (timeout)");
+};
+
+const main = async () => {
+  await deployContract();
+  let loopCount = 2;
+  while (loopCount > 0) {
+    await callContract(loopCount);
+    loopCount *= 2;
+  }
```

Comment:
This is an infinite loop right? Is that intentional?

Reply:
yes, the idea is to let it run until it errors

Reply:
Yes if, the node would support it, the loop would fail with overflow error.
Also, I didn't really do this script to be included in our repo, it was mostly to share the script (I made the PR to add description to it)